cargo-lock/src/package/source.rs

```diff
@@ -32,19 +32,53 @@ pub struct SourceId {
     /// The source kind.
     kind: SourceKind,
 
-    /// For example, the exact Git revision of the specified branch for a Git Source.
-    precise: Option<String>,
-
```

Comment:
Why was this removed? It still exists upstream: https://github.com/rust-lang/cargo/blob/1ce8023/src/cargo/core/source_id.rs#L56

Reply:
Basically, I was not aware that the code was copied from Cargo at some point, and from reading the code (and running the tests), it seemed to only be used/relevant in the git case. I could probably revert it. Do you have any other feedback I should process?

Reply:
The git case may be relevant to other users of `cargo-lock`.

In general that particular file is vendored from cargo (it's noted at the top, although it seems the URL needs to be updated)

Reply:
It seems to me that the `SourceId` and `SourceKind` stuff in Cargo has gotten a lot more intertwined between 2020 and now, so I'm not sure it's worth doing this. As an alternative, Cargo's cargo-util-schemas crate will include lockfile support in a few releases, so maybe we should drop cargo-lock in favor of using that directly.

Reply:
Looks like it provides `SourceKind` but not `SourceId`, which doesn't quite seem useful enough to start roping it in, especially since it seems to have breaking changes every few months